src/libstd/fs.rs

```diff
@@ -1572,6 +1606,12 @@ pub fn read_dir<P: AsRef<Path>>(path: P) -> io::Result<ReadDir> {
 
 /// Changes the permissions found on a file or a directory.
 ///
+/// Note: When you already have an open [`File`] object for the file in question,
+/// prefer [`File::set_permissions`].
+///
+/// [`File`]: struct.File.html
+/// [`File::set_permissions`]: struct.File.html#method.set_permissions
```

Comment:
Before you merge, would you like me to keep this note or wait till `File::set_permissions` stabilizes.